core/modules/layout/layout.admin.inc

```diff
@@ -2297,3 +2297,90 @@ function layout_ajax_form_update($form, $form_state) {
 
   return array('#type' => 'ajax', '#commands' => $commands);
 }
+
+/**
+ * Menu callback; Show the region styles form.
+ */
+function layout_configure_region_page($form, &$form_state, Layout $layout, $renderer_name, $region_name, $block_key = NULL) {
+  $form_state['layout'] = $layout;
+  $form_state['region_name'] = $region_name;
+  $layout_info = layout_get_layout_info($layout->layout);
+  if (!array_key_exists($region_name, $layout_info['regions'])) {
+    return MENU_NOT_FOUND;
+  }
+
+  $form['#tree'] = TRUE;
+
+  $styles = _layout_get_all_info('layout_style');
+  $style_options = array();
+  foreach ($styles as $style_name => $style_info) {
+    if (!empty($style_info['region theme'])) {
+      $style_options[$style_name] = $style_info['title'];
+    }
+  }
+  $current_style = 'region';
+  if (isset($form_state['values']['style'])) {
+    $current_style = $form_state['values']['style'];
+  }
+  if (count($style_options) > 1) {
+    $form['style'] = array(
+      '#type' => 'select',
+      '#title' => t('Style'),
+      '#options' => $style_options,
+      '#default_value' => $current_style,
+      '#ajax' => array(
+        'wrapper' => 'region_style_settings',
+        'callback' => 'layout_region_configure_ajax_style'
+      ),
+      '#parents' => array('style'),
+    );
+  }
+  else {
+    $form['style'] = array (
+      '#type' => 'value',
+      '#value' => 'region',
+    );
+  }
+  $form['style_settings'] = array(
+    '#type' => 'container',
+    '#id' => 'region_style_settings',
+    '#parents' => array('style_settings'),
+  );
+
+  $style = layout_create_handler('layout_style', 'region');
+  $style->form($form['style_settings'], $form_state);
+  $form_state['style'] = $style;
+
+  $form['actions'] = array('#type' => 'actions');
+  $form['actions']['submit'] = array(
+    '#type' => 'submit',
+    '#value' => t('Save styles'),
```

Comment:
Let's label this something more intuitive for users who won't recognize this as a "layout style", maybe just "Save configuration" to match the block dialog?